azuread/table_azuread_service_principal.go

```diff
@@ -19,159 +23,272 @@ func tableAzureAdServicePrincipal() *plugin.Table {
 		Name:        "azuread_service_principal",
 		Description: "Represents an Azure Active Directory (Azure AD) service principal",
 		Get: &plugin.GetConfig{
-			Hydrate:           getAdServicePrincipal,
-			ShouldIgnoreError: isNotFoundErrorPredicate([]string{"Invalid object identifier"}),
-			KeyColumns:        plugin.SingleColumn("id"),
+			Hydrate: getAdServicePrincipal,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isIgnorableErrorPredicate([]string{"Request_ResourceNotFound", "Invalid object identifier"}),
+			},
+			KeyColumns: plugin.SingleColumn("id"),
 		},
 		List: &plugin.ListConfig{
-			Hydrate:           listAdServicePrincipals,
-			ShouldIgnoreError: isNotFoundErrorPredicate([]string{"Request_UnsupportedQuery"}),
+			Hydrate: listAdServicePrincipals,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isIgnorableErrorPredicate([]string{"Request_UnsupportedQuery"}),
+			},
 			KeyColumns: plugin.KeyColumnSlice{
 				// Key fields
 				{Name: "display_name", Require: plugin.Optional},
-				{Name: "account_enabled", Require: plugin.Optional},
+				{Name: "account_enabled", Require: plugin.Optional, Operators: []string{"<>", "="}},
```

Comment:
Since we're explicitly defining the `Operators` now, what qual operators are enabled by default on key columns? Is it normally just `=`?

Reply:
Yes, @cbruno10, if we don't explicitly define any `Operators`, it automatically adds a single equal (`=`).
https://github.com/turbot/steampipe-plugin-sdk/blob/main/plugin/key_column.go#L103